agents/hybrid.py

```diff
@@ -132,6 +131,9 @@ def update_beliefs(self, current_state, action, reward, next_state, terminal):
             self.eligibility_traces = np.zeros((len(self.state_space), len(self.action_space)))
         else:
             self.previous_action = action
+        
+        # Decay all eligibility traces
+        self.eligibility_traces *= self._lambda
```

Comment:
why are you decaying this at the end? in stage 1 this doesn't matter as there is no lambda update. but on stage 2 this will first clear all eligibility traces to 0 then multiply it with lambda which doesn't make sense.

Reply:
I meant to decay it after the updates.
I think the effect is the same tho, I think it's usually done after the update, ( we either decay on zero in the beginning and then set one or set to zero then decay ) I will put it how it was

Reply:
do you agree on self.eligibility_traces[current_state, action] = 1 instead of += 1 ?